### ISSUE TITLE
StackOverflow when trying to get bit value from result set

### ISSUE THREAD

**Issue by kornilova203** [aws/amazon-redshift-jdbc-driver#8]
## Driver version
`2.0.0.1`

## Redshift version
`PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.22575`

## Client Operating System
MacOS

## JAVA/JVM version
1.8

## Problem description
```java
Statement statement = connection.createStatement();
statement.execute("SELECT '0'::bit");
ResultSet resultSet = statement.getResultSet();
resultSet.next();
System.out.println(resultSet.getBoolean(1)); // or getObject
```
**Expected result**
false
**Actual result**
```
Exception in thread "main" java.lang.StackOverflowError
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:206)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
	at com.amazon.redshift.jdbc.RedshiftResultSet.internalGetObject(RedshiftResultSet.java:209)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getObject(RedshiftResultSet.java:2898)
	at com.amazon.redshift.jdbc.RedshiftResultSet.getBoolean(RedshiftResultSet.java:2257)
```

**Comment by iggarish** [comment]
Thanks for reporting issue. We will look into it and get back to you.

**Comment by iggarish** [comment]
We found the root cause of this issue. We will release the fix in our next release, which is planned to release soon.

**Comment by kornilova203** [comment]
Cool! Thanks!

**Comment by iggarish** [comment]
Fix is available in 2.0.0.2 release.
